spring-cloud-netflix-zuul/src/main/java/org/springframework/cloud/netflix/zuul/ZuulProxyAutoConfiguration.java

```diff
@@ -89,15 +89,19 @@ public HasFeatures zuulFeature() {
 	@Bean
 	@ConditionalOnMissingBean(DiscoveryClientRouteLocator.class)
 	public DiscoveryClientRouteLocator discoveryRouteLocator() {
-		return new DiscoveryClientRouteLocator(this.server.getServlet().getContextPath(), this.discovery, this.zuulProperties,
+		String contextPath = this.server.getServlet().getContextPath();
+		if(contextPath == null) contextPath = "";
+		return new DiscoveryClientRouteLocator(contextPath, this.discovery, this.zuulProperties,
 				this.serviceRouteMapper, this.registration);
 	}
 
 	// pre filters
 	@Bean
 	@ConditionalOnMissingBean(PreDecorationFilter.class)
 	public PreDecorationFilter preDecorationFilter(RouteLocator routeLocator, ProxyRequestHelper proxyRequestHelper) {
-		return new PreDecorationFilter(routeLocator, this.server.getServlet().getContextPath(), this.zuulProperties,
+		String contextPath = this.server.getServlet().getContextPath();
+		if(contextPath == null) contextPath = "";
+		return new PreDecorationFilter(routeLocator, contextPath, this.zuulProperties,
```

Comment:
This should be removed and a `StringUtils.hasText()` added here https://github.com/spring-cloud/spring-cloud-netflix/blob/master/spring-cloud-netflix-zuul/src/main/java/org/springframework/cloud/netflix/zuul/filters/pre/PreDecorationFilter.java#L181